transformations/transformer_replace/transformation.py

```diff
@@ -0,0 +1,161 @@
+    ) -> Tuple[List[str], List[OriginalWord]]:
+        masked_texts = []
+        original_words = []
+        for token in doc:
+            if token.pos_ in self.pos_tokens and self._replace_tokens():
+                masked_texts.append(
+                    (
+                        doc[: token.i].text
+                        + " "
+                        + self.fill_pipeline.tokenizer.mask_token
+                        + " "
+                        + doc[token.i + 1 :].text
+                    ).strip()
+                )
+                original_words.append(OriginalWord(token.i, token.text))
+        return (masked_texts, original_words)
+
+    def generate_from_predictions(
+        self,
+        doc: spacy.tokens.Doc,
+        predictions: List[Any],
+        original_words: List[OriginalWord],
+    ) -> str:
+        if len(original_words) < 1:
+            return doc.text
+        elif len(original_words) == 1:
+            predictions = [predictions]
+
+        # initial text from beginning of sentence to first word to replace
+        new_text = doc[: original_words[0].index].text
+        for i, (preds, original_word) in enumerate(zip(predictions, original_words)):
+            predicted_scores = []
+            predicted_words = []
+            for p in preds:
+                if p["token_str"].strip() != original_word.text:
+                    predicted_scores.append(p["score"])
+                    predicted_words.append(p["token_str"])
+
+            # predict from scores
+            if self.sample_top_k:
+                x = torch.nn.functional.softmax(torch.tensor(predicted_scores), dim=0)
+                cat_dist = torch.distributions.categorical.Categorical(x)
+                selected_predicted_index = cat_dist.sample().item()
+            else:
+                selected_predicted_index = torch.tensor(predicted_scores).argmax()
+
+            new_text += predicted_words[selected_predicted_index]
+
+            # if there is another word to replace - append text in-between
+            if i + 1 < len(original_words):
+                new_text += (
+                    " "
+                    + doc[
+                        original_words[i].index + 1 : original_words[i + 1].index
+                    ].text
+                )
+            # else if last word then complete sentence
+            elif i + 1 == len(original_words):
+                new_text += " " + doc[original_words[i].index + 1 :].text
+        return new_text
+
+    def generate(self, sentence: str) -> List[str]:
+        doc = self.nlp(sentence)
+        masked_texts, original_words = self.get_masked_sentences_from_sentence(doc)
+        if len(masked_texts) > 0:
+            predictions = self.fill_pipeline(masked_texts)
+            new_sentence = self.generate_from_predictions(
+                doc, predictions, original_words
+            )
+            return [new_sentence]
+        else:
+            return [sentence]
```

Comment:
Might be more reasonable if it's just `[]` here so we can filter out unchanged examples.